tests/components/min_max/test_sensor.py

```diff
@@ -14,329 +13,334 @@
+    assert entity_ids[2] == state.attributes.get("min_entity_id")
+    assert MAX_VALUE == state.attributes.get("max_value")
+    assert entity_ids[1] == state.attributes.get("max_entity_id")
+    assert MEAN == state.attributes.get("mean")
 
-    def test_median_sensor(self):
-        """Test the median sensor."""
-        config = {
-            "sensor": {
-                "platform": "min_max",
-                "name": "test_median",
-                "type": "median",
-                "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
-            }
+
+async def test_not_enough_sensor_value(hass):
+    """Test that there is nothing done if not enough values available."""
+    config = {
+        "sensor": {
+            "platform": "min_max",
+            "name": "test_max",
+            "type": "max",
+            "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
         }
+    }
 
-        assert setup_component(self.hass, "sensor", config)
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
 
-        entity_ids = config["sensor"]["entity_ids"]
+    entity_ids = config["sensor"]["entity_ids"]
 
-        for entity_id, value in dict(zip(entity_ids, self.values)).items():
-            self.hass.states.set(entity_id, value)
-            self.hass.block_till_done()
+    hass.states.async_set(entity_ids[0], STATE_UNKNOWN)
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("sensor.test_median")
+    state = hass.states.get("sensor.test_max")
+    assert STATE_UNKNOWN == state.state
+    assert state.attributes.get("min_entity_id") is None
+    assert state.attributes.get("min_value") is None
+    assert state.attributes.get("max_entity_id") is None
+    assert state.attributes.get("max_value") is None
+    assert state.attributes.get("median") is None
 
-        assert str(float(self.median)) == state.state
-        assert self.min == state.attributes.get("min_value")
-        assert entity_ids[2] == state.attributes.get("min_entity_id")
-        assert self.max == state.attributes.get("max_value")
-        assert entity_ids[1] == state.attributes.get("max_entity_id")
-        assert self.mean == state.attributes.get("mean")
+    hass.states.async_set(entity_ids[1], VALUES[1])
+    await hass.async_block_till_done()
 
-    def test_not_enough_sensor_value(self):
-        """Test that there is nothing done if not enough values available."""
-        config = {
-            "sensor": {
-                "platform": "min_max",
-                "name": "test_max",
-                "type": "max",
-                "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
-            }
-        }
+    state = hass.states.get("sensor.test_max")
+    assert STATE_UNKNOWN != state.state
+    assert entity_ids[1] == state.attributes.get("min_entity_id")
+    assert VALUES[1] == state.attributes.get("min_value")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)